opslevel/datasource_opslevel_repository.go

```diff
@@ -1,38 +1,99 @@
 package opslevel
 
-// import (
-// 	"github.com/opslevel/opslevel-go/v2024"
-
-// 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-// )
-
-// func datasourceRepository() *schema.Resource {
-// 	return &schema.Resource{
-// 		Read: wrap(datasourceRepositoryRead),
-// 		Schema: map[string]*schema.Schema{
-// 			"alias": {
-// 				Type:        schema.TypeString,
-// 				Description: "A human-friendly, unique identifier for the repository.",
-// 				ForceNew:    true,
-// 				Optional:    true,
-// 			},
-// 			"id": {
-// 				Type:        schema.TypeString,
-// 				Description: "The id of the repository to find.",
-// 				ForceNew:    true,
-// 				Optional:    true,
-// 			},
-// 		},
-// 	}
-// }
-
-// func datasourceRepositoryRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	resource, err := findRepository("alias", "id", d, client)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	d.SetId(string(resource.Id))
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure RepositoryDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &RepositoryDataSource{}
+
+func NewRepositoryDataSource() datasource.DataSource {
+	return &RepositoryDataSource{}
+}
+
+// RepositoryDataSource manages a Repository data source.
+type RepositoryDataSource struct {
+	CommonDataSourceClient
+}
+
+// RepositoryDataSourceModel describes the data source data model.
+type RepositoryDataSourceModel struct {
+	Alias      types.String `tfsdk:"alias"`
+	Id         types.String `tfsdk:"id"`
+	Identifier types.String `tfsdk:"identifier"`
+	Name       types.String `tfsdk:"name"`
+}
+
+func NewRepositoryDataSourceModel(ctx context.Context, repository opslevel.Repository, identifier string) RepositoryDataSourceModel {
+	return RepositoryDataSourceModel{
+		Alias:      types.StringValue(repository.DefaultAlias),
+		Id:         types.StringValue(string(repository.Id)),
+		Identifier: types.StringValue(identifier),
+		Name:       types.StringValue(repository.Name),
+	}
+}
+
+func (d *RepositoryDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_repository"
+}
+
+func (d *RepositoryDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
```

Comment:
❓ This schema is doesn't match [the public docs](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/data-sources/repository) or the old code